src/lib/sedlexing.ml

```diff
@@ -492,15 +496,15 @@ module Utf16 = struct
         Buffer.add_char buf c4
        )
 
-    let from_int_array bo a apos len bom =
-      let b = Buffer.create (len * 4) in
+    let from_uchar_array bo a apos len bom =
+      let b = Buffer.create (len * 4 + 2) in
```

Comment:
`+2` ?

Reply:
Ah, technically not necessary, but the Buffer.create call wasn't allocating space for the Byte Order Mark. It will, of course, automatically allocate more on its own if needed, but as long as we were allocating enough for all possible four byte values it seemed like allocating enough for the BOM was good, too. Was this silly? I suspect there's no right answer.

Reply:
Ah, technically not necessary, but the Buffer.create call wasn't allocating space for the Byte Order Mark. It will, of course, automatically allocate more on its own if needed, but as long as we were allocating enough for all possible four byte values it seemed like allocating enough for the BOM was good, too. Was this silly? I suspect there's no right answer.

Reply:
Right, that makes sense.